src/main/java/org/openrewrite/maven/MavenMojoProjectParser.java

```diff
@@ -198,12 +198,16 @@ public Stream<SourceFile> listSourceFiles(MavenProject mavenProject, @Nullable X
             return sourceFile;
         }).filter(Objects::nonNull);
 
+        // Use runtime Java version for additional resources, as they do not explicitly use a source/target version.
+        List<Marker> additionalResourcesProvenance = new ArrayList<>(projectProvenance);
+        additionalResourcesProvenance.add(getJavaVersionMarker(null, null));
```

Comment:
Instead of the runtime Java version, we could also opt to go for the src/main JavaVersion marker here too; not correct, but could be less surprising. In any case I'd expect it to be rare that these resources even need a JavaVersion marker, so this is mostly in case anyone built something we didn't foresee here, as we were surprised before with markers affecting wrappers for instance.

Reply:
I don't think adding the `JavaVersion` marker outside of the source set is correct. As a parallel from the Gradle plugin, over there we don't add a `JavaVersion` marker to anything that is outside of the source set either.

Reply:
Even better; I'd kept this out of caution, but if you also feel the markers should not be there I'll remove them.

Reply:
Picked up in https://github.com/openrewrite/rewrite-maven-plugin/pull/729/commits/2a2d1f625843bd871b20b6ebc2e21d386159a58b